### PR TITLE
Feat/download as text button

### LINKED ISSUE #20: Download generated bylaws as text
## Value statement - a requirement, not a solution
As a user
I need to download the generated bylaws
so that I can save them easily.

## Description (Text, Scribble, Wireframe, Design)
Screenshot 2020-09-28 at 12 04 12

## Acceptance criteria
- [x] user can download generated bylaws as txt
- [x] button text changes when user clicked button

## Tasks
- [x] create feature branch
- [x] add download button
- [x] add download logic

## Size
medium

### PR DESCRIPTION
**Pull request by carolinwagner**
Please review :-) 

Preview: https://capstone-project-git-feat-download-as-text-button.carolinwagner.vercel.app

Closes #20

### CODE REVIEW

File: src/BylawsText.js [hunk 1]
```diff
@@ -37,6 +42,22 @@ export default function BylawsText({ answers }) {
         .then(() => setIsTextCopied(true))
   }
 
+  const downloadBylaws = (text, fileType, fileName) => {
```
Comment (karstenlenz) [inline src/BylawsText.js:45]: Is there a reason why don't use a function declaration like  `function downloadByLaws(test, fileType, fileName){...}` ?
I think it makes it a little bit easier to see what is a variable and what's a function. Also, if you use function declarations you don't need to worry about the order of the code and you can put all the functions at the bottom, if you like.

File: src/BylawsText.js [hunk 2]
```diff
@@ -37,6 +42,22 @@ export default function BylawsText({ answers }) {
...
+    const blob = new Blob([text], { type: fileType })
+
+    const a = document.createElement('a')
```
Comment (karstenlenz) [inline src/BylawsText.js:48]: I'm not totally sure about this, but I think using `document.createElement` in React is unusal. Normally you let React create all the elements for you.

File: src/BylawsText.js [hunk 3]
```diff
@@ -61,11 +82,26 @@ export default function BylawsText({ answers }) {
...
               <ClipboardIcon />
               <StyledCopyText>
-                Satzungstext {isTextCopied ? 'kopiert' : 'kopieren'}
+                Satzungstext {isTextCopied ? 'wurde kopiert' : 'kopieren'}
               </StyledCopyText>
             </Button>
+            <Button
+              variant="secondary"
+              onClick={() => {
+                const text = bylawsRef.current?.innerText
```
Comment (karstenlenz) [inline src/BylawsText.js:91]: This looks like you get the download-text from the HTML.innertext. Is that necessary? Don't you have the ByLaws content in a state or variable somewhere?

File: src/BylawsText.js [hunk 4]
```diff
@@ -111,8 +149,19 @@ const StyledContainer = styled.section`
 
 const StyledSmallButtonContainer = styled.div`
   display: flex;
+  gap: 20px;
```
Comment (karstenlenz) [inline src/BylawsText.js:152]: The gap property is not supported by all major browsers yet. You might want to look into some other way to position the items, like some margin for example.

see here https://caniuse.com/flexbox-gap

**Review by karstenlenz — APPROVED** [review]
Looks great! I didn't understand everything, so I left some questions. If you think everything is fine, you can feel free to ignore them :)

File: src/BylawsText.js (hunk 1 above)
Comment (carolinwagner) [reply src/BylawsText.js:45]: Good point :-) I changed it into a function declaration

File: src/BylawsText.js (hunk 4 above)
Comment (carolinwagner) [reply src/BylawsText.js:152]: I changed it :-)

File: src/BylawsText.js (hunk 3 above)
Comment (carolinwagner) [reply src/BylawsText.js:91]: I am using the innertext attribute to get rid of all the html formatting within the generated Bylaws text. Also the Bylaws text consists of many variables and I don't have a single variable where everything is stored.

File: src/BylawsText.js (hunk 2 above)
Comment (carolinwagner) [reply src/BylawsText.js:48]: You're right, but I coudn't do it with React. This was my source: https://gist.github.com/danallison/3ec9d5314788b337b682

**Pull request merged**